From 2cea3c03f0026740780b2602a921fa417a5c2e30 Mon Sep 17 00:00:00 2001
From: Huang Rui <ray.huang@amd.com>
Date: Tue, 14 Feb 2023 14:38:46 +0800
Subject: [PATCH 14/49] swiotlb-xen: expose xen_phys_to_dma and
 range_straddles_page_boundary for future use

These functions will be used for phy dma ops.

Signed-off-by: Huang Rui <ray.huang@amd.com>
---
 drivers/xen/swiotlb-xen.c | 4 ++--
 include/xen/swiotlb-xen.h | 3 +++
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/xen/swiotlb-xen.c b/drivers/xen/swiotlb-xen.c
index 0e6c6c25d154..e2199456e0c8 100644
--- a/drivers/xen/swiotlb-xen.c
+++ b/drivers/xen/swiotlb-xen.c
@@ -53,7 +53,7 @@ static inline phys_addr_t xen_phys_to_bus(struct device *dev, phys_addr_t paddr)
 	return baddr;
 }
 
-static inline dma_addr_t xen_phys_to_dma(struct device *dev, phys_addr_t paddr)
+dma_addr_t xen_phys_to_dma(struct device *dev, phys_addr_t paddr)
 {
 	return phys_to_dma(dev, xen_phys_to_bus(dev, paddr));
 }
@@ -74,7 +74,7 @@ static inline phys_addr_t xen_dma_to_phys(struct device *dev,
 	return xen_bus_to_phys(dev, dma_to_phys(dev, dma_addr));
 }
 
-static inline int range_straddles_page_boundary(phys_addr_t p, size_t size)
+int range_straddles_page_boundary(phys_addr_t p, size_t size)
 {
 	unsigned long next_bfn, xen_pfn = XEN_PFN_DOWN(p);
 	unsigned int i, nr_pages = XEN_PFN_UP(xen_offset_in_page(p) + size);
diff --git a/include/xen/swiotlb-xen.h b/include/xen/swiotlb-xen.h
index 808d17ad8d57..99063b9ce7b6 100644
--- a/include/xen/swiotlb-xen.h
+++ b/include/xen/swiotlb-xen.h
@@ -12,4 +12,7 @@ void xen_dma_sync_for_device(struct device *dev, dma_addr_t handle,
 
 extern const struct dma_map_ops xen_swiotlb_dma_ops;
 
+dma_addr_t xen_phys_to_dma(struct device *dev, phys_addr_t paddr);
+int range_straddles_page_boundary(phys_addr_t p, size_t size);
+
 #endif /* __LINUX_SWIOTLB_XEN_H */
-- 
2.17.1

